### PR TITLE
Make AntFixture handling task provider api compliant

### PR DESCRIPTION
**Pull request by breskeby**
This tweaks the AntFixture handling to make it compliant with the task avoidance api.
Tasks of type `StandaloneRestTestTask` are now generally finalised by using the typed ant stop task 
which allows us to remove of errorprone `dependsOn` overrides in `StandaloneRestTestTask`. As a result
we also ported more task definitions in the build to task avoidance api. 

Next work item regarding AntFixture handling is porting `AntFixture` to a plain Gradle task and remove 
Groovy AntBuilder will allow us to port more build logic from Groovy to Java but is out of the scope of 
This PR.

### CODE REVIEW

**Comment by breskeby** [comment]
checkPart2 initial fails with `ExecuteWatchQueuedStatsTests` timing out. Seen that before occasionally before https://gradle-enterprise.elastic.co/s/irvon7jqsdcs6/tests/:x-pack:plugin:watcher:internalClusterTest/org.elasticsearch.xpack.watcher.execution.ExecuteWatchQueuedStatsTests/testQueuedStats#1 I don't think this PR had any specific impact on that test

**Comment by breskeby** [comment]
@elasticmachine test this please

File: buildSrc/src/main/groovy/org/elasticsearch/gradle/AntFixtureStop.groovy [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    @Inject
+    AntFixtureStop(FileSystemOperations fileSystemOperations) {
+        super(fileSystemOperations)
+        this.fileSystemOperations = fileSystemOperations
+    }
+
+    void setFixture(AntFixture fixture) {
+        assert this.fixture == null
+        this.fixture = fixture;
+        final Object pid = "${ -> this.fixture.pid }"
```
Comment (pugnascotia) [inline buildSrc/src/main/groovy/org/elasticsearch/gradle/AntFixtureStop.groovy:46]: My Groovy knowledge is weak - does this result in a lazily-evaluated string that returns a PID?

File: plugins/discovery-ec2/qa/amazon-ec2/build.gradle [hunk 2]
```diff
@@ -64,61 +66,66 @@ yamlRestTest.enabled = false
...
+testClusters.matching { it.name == "yamlRestTestContainerCredentials" }.configureEach {
   environment 'AWS_CONTAINER_CREDENTIALS_FULL_URI',
     { "http://${-> tasks.findByName("ec2FixtureContainerCredentials").addressAndPort}/ecs_credentials_endpoint" }, IGNORE_VALUE
 }
 
 // Extra config for InstanceProfile
-ec2FixtureInstanceProfile.env 'ACTIVATE_INSTANCE_PROFILE', true
+tasks.named("ec2FixtureInstanceProfile").configure {
+  env 'ACTIVATE_INSTANCE_PROFILE', true
+}
```
Comment (pugnascotia) [inline plugins/discovery-ec2/qa/amazon-ec2/build.gradle:131]: Does your editor not insert a trailing newline?

**Review by pugnascotia — APPROVED** [review]

File: plugins/discovery-ec2/qa/amazon-ec2/build.gradle (hunk 2 above)
Comment (breskeby) [reply plugins/discovery-ec2/qa/amazon-ec2/build.gradle:131]: nope. should it?

File: buildSrc/src/main/groovy/org/elasticsearch/gradle/AntFixtureStop.groovy (hunk 1 above)
Comment (breskeby) [reply buildSrc/src/main/groovy/org/elasticsearch/gradle/AntFixtureStop.groovy:46]: exactly. Ultimately we want these tasks to be java and not rely on those groovy "hacks", but porting AntFixture requires a bit more work to remove the `ant` bit.

**Pull request merged**